src/Support/DataType.php

```diff
@@ -22,6 +22,10 @@
 
 class DataType implements Countable
 {
+    private const READONLY_BITMASK = 128;
```

Comment:
I don't think this is the right place to put readonly. A type is technically not readonly, a property is. So let's put it over there (DataProperty). Or am I missing something?